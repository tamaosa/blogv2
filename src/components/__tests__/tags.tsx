import React from "react"
import { render } from "@testing-library/react"
import Tags from "../tags"

describe("Tags", () => {
  it("snapshot test1", () => {
    const { asFragment } = render(<Tags />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("snapshot test2", () => {
    const { asFragment } = render(
      <Tags tags={["tag1", "tag2", "tag3", "tag4", "tag5"]} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("matching tag", () => {
    const { container } = render(
      <Tags tags={["tag1", "tag2", "tag3", "tag4", "tag5"]} />
    )
    expect(container).toHaveTextContent("#tag1#tag2#tag3#tag4#tag5")
  })

  it("testing links to tags", () => {
    const tags = ["tag1", "tag2", "tag3", "tag4", "tag5"]
    const { container } = render(<Tags tags={tags} />)
    for (let tag of tags) {
      expect(container.innerHTML).toMatch(
        `<a aria-label="タグへのリンク" href="/tags/${tag}/">#${tag}</a>`
      )
    }
  })
})
