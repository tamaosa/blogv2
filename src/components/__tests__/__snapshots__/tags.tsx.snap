// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLink snapshot test1 1`] = `
<DocumentFragment>
  <div>
    <ul
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
  </div>
</DocumentFragment>
`;

exports[`ArticleLink snapshot test2 1`] = `
<DocumentFragment>
  <div>
    <ul
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <li>
        <a
          href="/tags/tag1/"
        >
          #tag1
        </a>
      </li>
      <li>
        <a
          href="/tags/tag2/"
        >
          #tag2
        </a>
      </li>
      <li>
        <a
          href="/tags/tag3/"
        >
          #tag3
        </a>
      </li>
      <li>
        <a
          href="/tags/tag4/"
        >
          #tag4
        </a>
      </li>
      <li>
        <a
          href="/tags/tag5/"
        >
          #tag5
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
